lightning/src/ln/functional_test_utils.rs

```diff
@@ -2353,14 +2353,15 @@ macro_rules! expect_payment_path_successful {
 
 /// Returns the total fee earned by this HTLC forward, in msat.
 pub fn expect_payment_forwarded<CM: AChannelManager, H: NodeHolder<CM=CM>>(
-	event: Event, node: &H, prev_node: &H, next_node: &H, expected_fee: Option<u64>,
```

Comment:
Thanks for pointing that out—makes sense!  
My initial reasoning was that since `next_node.node().get_our_node_id()` and `next_node_id` are practically equal, a single check seemed sufficient.  
But I see your point—keeping both checks ensures we validate two different equivalences.
Updating! 🧑‍💻

Reply:
Updated in [pr3458.02](https://github.com/shaavan/rust-lightning/commits/pr3458.02)
Thanks for the pointer! ✨